pyquil/api/tests/test_quantum_computer.py

```diff
@@ -205,3 +205,43 @@ def test_qc_joint_calibration(forest):
     # Z1 expectation value for state |01> with 95% RO fid on both qubits is about 0.9
     assert np.isclose(results[0].additional_results[1].expectation, 0.9, atol=0.01)
     assert results[0].additional_results[1].total_counts == 40000
+
+
+def test_qc_expectation_on_qvm_that_requires_executable(forest):
+    # regression test for https://github.com/rigetti/forest-tutorials/issues/2
+    device = NxDevice(nx.complete_graph(2))
+    qc = QuantumComputer(
+        name="testy!",
+        qam=QVM(connection=forest, requires_executable=True),
+        device=device,
+        compiler=DummyCompiler(),
+    )
+
+    p = Program()
+    theta = p.declare("theta", "REAL")
+    p += RESET()
+    p += RY(theta, 0)
+    p.wrap_in_numshots_loop(10000)
+
+    sx = ExperimentSetting(in_state=sZ(0), out_operator=sX(0))
+    e = Experiment(settings=[sx], program=p)
+
+    thetas = [-np.pi / 2, 0.0, np.pi / 2]
+    results = []
+
+    # Here we verify that multiple calls to qc.experiment with the same experiment backed by a QVM
+    # that requires_exectutable does not raise an exception.
+    for theta in thetas:
+        results.append(qc.experiment(e, memory_map={"theta": [theta]}))
+
+    assert np.isclose(results[0][0].expectation, -1.0, atol=0.01)
+    assert np.isclose(results[0][0].std_err, 0)
+    assert results[0][0].total_counts == 20000
+
+    assert np.isclose(results[1][0].expectation, 0.0, atol=0.1)
+    assert results[1][0].std_err < 0.01
```

Comment:
Unclear if these are reasonable values for `atol` and `std_err` here or likely to cause flakiness.

Working through exercise 2.59 in Nielsen & Chuang, I find that given a qubit in the state `|0>`, then the average value of the observable X should be 0 with stddev = 1.

Running the test `RY(theta) 0` for 10,000 shots with theta = 0 results in an expectation of 0.0057 with `std_err` of ~ 0.007. According to [wikipedia](https://en.wikipedia.org/wiki/Standard_error#Estimate) the standard error is estimated like `s / sqrt(n)` where s is the sample stddev and n is number of samples, which seems to be at least in the ballpark of agreement.

If I run the test in a loop 100 times, the `std_err` is always between 0.007069 and 0.007072.

Reply:
Other than the intellectual exercise, it's not that important.  I believe the more sophisticated tests in `test_operator_estimation` would catch something truly awry.  Maybe leave a comment explaining your choices.

Reply:
left a small comment in the latest commit.

Reply:
speaking of flaky tests:

https://travis-ci.org/github/rigetti/pyquil/jobs/661211565#L1258

Reply:
nice